runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java

```diff
@@ -297,8 +317,8 @@ public Value readGlobal(int idx) {
             return imports.global(idx).instance().getValue();
         }
         var i = idx - importedGlobalsOffset;
-        if (i < 0 || i >= globals.length) {
-            throw new InvalidException("unknown global " + i);
+        if (i < 0 || i >= globals.length || globals[idx - importedGlobalsOffset] == null) {
```

Comment:
How do we have a null global for a valid array index?

Reply:
I just get nervous seeing nulls in unexpected places. If this is expected, then no problem

Reply:
It's actually a good catch: https://github.com/dylibso/chicory/pull/421/commits/9a751db8b2df27ae1547b013dc65913bf699f60f

Thanks!